Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contest/ContestPage.tsx

```diff
@@ -97,11 +98,13 @@ const ContestPage = () => {
         [ contestError, renderErrorMessage ],
     );
 
-    const renderPage = useMemo(
-        () => isParticipationTypeValid(participationType)
-            ? renderContestPage
-            : <HomePage />,
-        [ participationType, renderContestPage ],
+    useEffect(
+        () => {
+            if (!isNil(participationType) && !isParticipationTypeValid(participationType)) {
```

Comment:
Shouldnt we navigate to home if the contest is not found also?

Reply:
There is an error displaying a message that the contest is not found.